Sources/UAPusher/Provider.swift

```diff
@@ -1,20 +1,27 @@
+import TLS
 import Vapor
 
 public final class Provider: Vapor.Provider {
     let config: UAPusherConfig
     public static var repositoryName: String = "uapusher"
     private var connectionManager: ConnectionManager
 
-    public func boot(_ config: Config) throws {}
+    public func boot(_ config: Vapor.Config) throws {}
 
     public func boot(_ drop: Droplet) throws {
+        connectionManager.client = try drop.config.resolveClient().makeClient(
```

Comment:
What's your thoughts around doing this vs. init'en the `ConnectionManager` in `boot`?

Reply:
I'll give it a try, see how that compares

Reply:
The `ConnectionManager` requires the config. We could get it from the `Droplet`.

Reply:
We should have the config on this Provider - or am I misunderstanding something?

Reply:
No, you understand correctly. I retracted my comment already :P I was confused because different types are called `config`.

Reply:
We do still need the `Droplet`'s config for resolving the `ClientFactory` though:

```
    public func boot(_ drop: Droplet) throws {
        let connectionManager = try ConnectionManager(
            config: self.config,
            clientFactory: drop.config.resolveClient()
        )

        drop.uapusher = UAManager(connectionManager: connectionManager)
    }
```